### PR TITLE
Update mergesort.py

### PR DESCRIPTION
**Pull request by akashgkrishnan**
1) Updating the merge sort in python as the previous implementation was modifying the input array
2) divided the division part and conquer part of the merge sort algorithm as 2 functions namely mergeSort and merge.
3) function mergeSort divides the function into halves i.e the purpose of the function will be to divide the array
4) function merge will merge 2 halves into a sorted array
5) Added test case for the base case
6
The time and space complexity of the previous and my version remains the same
i.e (n log(n)  time and n log(n) space

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: divide_and_conquer/mergesort.py [hunk 1]
```diff
@@ -1,48 +1,53 @@
...
-        k += 1
-    while j < len(right_arr):
-        arr[k] = right_arr[j]
-        j += 1
-        k += 1
-    return arr
-
-
-def mergesort(arr, left, right):
+def mergeSort(array):
```
Comment (dhruvmanila) [inline divide_and_conquer/mergesort.py:1]: The function name and variable names should be in `snake_case` as mentioned in CONTRIBUTING.md. Please change all of them accordingly.

File: divide_and_conquer/mergesort.py [hunk 2]
```diff
@@ -1,48 +1,53 @@
...
+    # Split the array into halves till the array length becomes equal to One
+    # merge the arrays of single length returned by mergeSort function and
+    # pass them into the merge arrays function which merges the array
+    leftHalf = array[:middle]
+    rightHalf = array[middle:]
+
+    return mergeArrays(mergeSort(leftHalf), mergeSort(rightHalf))
+
+
+def mergeArrays(leftHalf, rightHalf):
```
Comment (dhruvmanila) [inline divide_and_conquer/mergesort.py:24]: `snake_case`, type hints, `doctest`

File: divide_and_conquer/mergesort.py [hunk 3]
```diff
@@ -1,48 +1,53 @@
...
+def mergeSort(array):
     """
-    >>> mergesort([3, 2, 1], 0, 2)
+    >>> mergeSort([3, 2, 1])
     [1, 2, 3]
-    >>> mergesort([3, 2, 1, 0, 1, 2, 3, 5, 4], 0, 8)
+    >>> mergeSort([3, 2, 1, 0, 1, 2, 3, 5, 4])
     [0, 1, 1, 2, 2, 3, 3, 4, 5]
+    >>> mergeSort([10])
+    [10]
```
Comment (dhruvmanila) [inline divide_and_conquer/mergesort.py:8]: Please add random tests in here.
Use `from random import shuffle`

**Comment by akashgkrishnan** [comment]
closed the PR due to some issues in the PEP 8 std and some other build related issues

**Pull request closed without merging**